### DISCUSSION TITLE
회고

### DISCUSSION THREAD

**Discussion by kimchijinju** [Webwaiter/Webwaiter#119] [General]
### 무엇을 해냈는가?
1. 시간내에 요구사항에 맞춰 작동하도록 프로그램을 만들었다.

### 무엇이 부족했나?
2. 객체지향적으로 프로그램을 만들고 싶었으나, 객체지향적으로 짠다는 것이 뭔지 잘 몰랐기때문에 부족했다.

### 좋았던 점
1. 전체적인 흐름을 생각해보고 프로그램 전체 흐름 안에서 객체가 가질 수 있는 상태를 정하여 관리하였다.
2. 프로토타입을 빠르게 만들어봐서 프로젝트의 전체적인 흐름을 파악하기 좋았다.
3. 설계하며 모르는 것을 빠르게 학습하며 진행했다.
4. 팀원간의 의사소통이 활발하여 협업에 무리가 없었다.

### 개선해야 할 점
1. 깃 관리
I. main브랜치가 항상 실행 가능한 상태가 아니었다.
II. 프로젝트 디렉토리 구조를 먼저 정하고 시작하지 못했다.
III. 이슈, pr 내용 작성의 컨벤션 , 템플릿등 규칙을 정했으면 좋았을 것 같다.
2. 테스트 과정
I. 코드를 수정하고 나서 테스트를 하는데 어려웠다.
II. 테스트 시나리오를 먼저 만들고, 더 잘 할 수 있다면 관련 tool를 사용해서 자동화 했으면 좋았을 것 같다.
3. 개발 프로세스
I. 실제 개발 프로세스가 어떻게 이뤄지는지 파악할 수 있었으면 좋았을 것 같다.
II. 요구사항 분석을 좀더 확실하게 했으면 설계에 막힘이 없었을 것 같다.
III. 코드 외에 산출물이 잘 안나왔다.
4. 프로그램 기능 측면, 리펙토링 측면
I. 청크드 응답 메세지를 구현한다.
II. PUT 메서드를 구현하고 서브젝트에서 제공하는 테스터기를 사용해 테스트 해본다.
III. kqueue와 work queue를 합칠수 있다.
IV. 결론 : 코드를 한번 정리하면 좋을 것 같다.
4. 프로젝트 시간 관리, 학습 과정 효율화
I. 학습을 하면서 요구하는 테스트 케이스를 같이 짜봤으면 좋았을 것 같다.
II. 같이 모여서 집중적으로 학습했다면 학습시간이 줄었을 것이다.
### 앞으로 무엇을 하면 좋을까?
1. 객체지향에 대해 참고 할 수 있는 책이나, 레퍼런스 코드를 분석하여 객체지향에 대해 공부한다.
2. 시간에 맞춰 구현하다보니 코드가 정리되지 않았는데, 구조를 잘 정리하고 리팩토링 한다.
3. 개발 테스트에 대해 공부한다.
4. CI/CD에 대해 공부한다.

**Comment by flowjiyun** [comment]
이번 프로젝트에서 부족한 점을 다음 웹프로젝트에서는 꼭 개선해봅시다!!
